src/BPSecLib_Private.h

```diff
@@ -700,6 +701,14 @@ int BSL_SecParam_InitBytestr(BSL_SecParam_t *self, uint64_t param_id, BSL_Data_t
  */
 int BSL_SecParam_InitInt64(BSL_SecParam_t *self, uint64_t param_id, uint64_t value);
 
+/**
+ * @param self This Security Paramter
+ * @param param_id ID of the parameter
+ * @param value text string of the parameter
+ * @returns Negative on an error.
+ */
+int BSL_SecParam_InitStr(BSL_SecParam_t *self, uint64_t param_id, const char *value);
```

Comment:
Its good to annotate in vs in,out param especially for pointers. Also good to mention that this text string is copied from so no need for caller to retain.